python/agent-inbox/main.py

```diff
@@ -96,11 +95,13 @@ def start_agent(request: AgentStartRequest):
 async def list_waiting_agents():
     # List all active agents and retrieve their statuses
     agent_workflows = await DBOS.list_workflows_async(
-        status="PENDING", name=durable_agent.__qualname__
+        status="PENDING", name=durable_agent.__qualname__, app_version=DBOS.application_version,
```

Comment:
Why is this filter needed?

Reply:
This is so that old-version agents do not show up in the current-version UI for approval. Because, otherwise, they are not really actionable.

Reply:
Yes. This is core to the design. Did a lot of careful work in the recent fork-events PR to make sure it was backwards compatible, for example.